guard-lambda/src/main.rs

```diff
@@ -47,7 +93,38 @@ pub(crate) async fn call_cfn_guard(e: CustomEvent, _c: Context) -> Result<Custom
         let json_value: serde_json::Value = serde_json::from_str(&result)?;
         results_vec.push(json_value)
     }
+
+    //let response: CustomOutput;
+    let mut resp = Vec::new();
+
+    if e.s3_output_bucket.is_empty() {
+        resp = results_vec;
+    }else {
+        // No extra configuration is needed as long as your Lambda has 
+        // the necessary permissions attached to its role.
+        let config = aws_config::from_env().load().await;
+
+        // Create an S3 client
+        let client = aws_sdk_s3::Client::new(&config);
+
+        // Generate a filename based on when the request was received.
+        let filename = format!("{}.json", time::OffsetDateTime::now_utc().unix_timestamp());
```

Comment:
We should consider generating filename from timestamp as default behavior only if the related file name parameter aren't passed by the end-user. When a user passes file names as input, we should only append the timestamp to base prefix and suffix passed. We can rename the parameter itself to something like `s3_publisher` and change the datatype from string to instead a datatype containing all the dependent parameters like `bucket_name` (needs to be mandatory if we are supposed to be publishing to S3 i.e `if s3_publisher != None`),  `base_prefix` (optional, defaults to empty string) and `base_suffix` (optional, defaults to `.json`) , this would allow flexibility to user to call Lambda for different set of similar inputs and publish it with different file name patterns to the same bucket. 

Also, let's also make the timestamps more human-readable ones, using ISO 8601 format, with UTC timezone as follows.

```
use std::time::{SystemTime};
use chrono::prelude::{DateTime, Utc};

// method to initialize the timestamp string
fn iso8601(st: &std::time::SystemTime) -> String {
    let dt: DateTime<Utc> = st.clone().into();
    format!("{}", dt.format("%+"))
}

// call this in the filename above
println!("{}", iso8601(&SystemTime::now()));
// prints 2022-05-03T19:29:15.696460343+00:00
```

Example with all parameters defined, `bucket-name = 'my-output-bucket'` and `base-prefix = 'service1/result_'`
```
/*
** Format: 
** s3://<bucket-name>/<base-prefix><auto-generated-disambiguator><base-suffix>
*/
s3://my-output-bucket/service1/result_2022-05-03T19:29:15.696460343+00:00.json
```

When prefix and suffix aren't passed this would default to:

```
s3://my-output-bucket/2022-05-03T19:29:15.696460343+00:00.json
```


Example:
```
Service 1 Infrastructure reports may get published as:

s3://my-output-bucket/service1/result_2022-05-03T19:29:15.696460343+00:00.json
s3://my-output-bucket/service1/result_2022-05-03T19:29:15.696460343+00:00.json
s3://my-output-bucket/service1/result_2022-05-03T19:29:15.696460343+00:00.json

Service 2 Infrastructure reports may get published as:
s3://my-output-bucket/service2/result_2022-05-03T19:29:15.696460343+00:00.json
s3://my-output-bucket/service2/result_2022-05-03T19:29:15.696460343+00:00.json
s3://my-output-bucket/service2/result_2022-05-03T19:29:15.696460343+00:00.json
```